sycl/unittests/Extensions/DeviceArchitectureOneArchSelected.cpp

```diff
@@ -0,0 +1,41 @@
+//==--------------- DeviceArchitectureOneArchSelected.cpp ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <gtest/gtest.h>
+
+// define one of __SYCL_TARGET_INTEL_*** macro, e.g., the one for SKL
```

Comment:
I don't see why we have to have two tests for 2/1 cases. What are we trying to catch here?

Reply:
One test case checks single platform passed to -fsycl-targets, another one checks some kind of co-existence of multiple platforms passed to -fsycl-targets. We can't create a single test case for both of them because there are macros which should be compiled within separate translation units.
Overall, don't quite understand, can you please elaborate?

Reply:
Actually, removed DeviceArchitectureMultipleArchSelected.cpp, looking internally on it, it looks like duplication